python/pyspark/sql/session.py

```diff
@@ -508,6 +508,11 @@ def getOrCreate(self) -> "SparkSession":
 
                             if url is None and is_api_mode_connect:
                                 url = opts.get("spark.master", os.environ.get("MASTER", "local"))
+                                if url.startswith("sc://"):
```

Comment:
instead of a bandaid fix, can we refactor the code to have different branches for `--remote` and `--master`?

Reply:
The problem is that this is Python specific. Spark Connect Server is launched without Spark Submit itself. For other cases, the exceptions are covered via `SparkSubmit.prepareSubmitEnvironment`.